tools/src/launcher/Main.hs

```diff
@@ -515,11 +514,55 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
+runWallet
+    :: Bool                               -- ^ wallet logging
+    -> (FilePath, [Text], Maybe FilePath) -- ^ Wallet, its args, wallet log file
+    -> (FilePath, [Text], Maybe FilePath) -- ^ Node, its args, node log file
+    -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) (_,_, nLogPath) = do
     logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            cr <- createLogFileProc path args lp
+            system' phvar cr mempty EWallet
+        Nothing ->
+           -- if nLog is Nothing and shouldLog is True
+           -- we want to CreatePipe otherwise Inherit
+           let cr = if shouldLog && isNothing nLogPath then
+                        Process.CreatePipe
+                    else Process.Inherit
+           in system' phvar (createProc cr path args) mempty EWallet
+
+createLogFileProc :: FilePath -> [Text] -> FilePath -> IO Process.CreateProcess
+createLogFileProc path args lp = do
+    _ <- createDirectoryIfMissing True (directory lp)
+    (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+    IO.hSetBuffering logHandle IO.LineBuffering
+    return $ createProc (Process.UseHandle logHandle) path args
+
+createProc :: Process.StdStream -> FilePath -> [Text] -> Process.CreateProcess
+createProc stdStream path args =
+    (Process.proc path (map toString args))
+            { Process.std_in = Process.CreatePipe
+            , Process.std_out = stdStream
+            , Process.std_err = stdStream
+            }
+
+asyncLog :: Handle -> Handle -> ProcessHandle -> Executable -> IO ExitCode
+asyncLog stdO stdE pH nt = do
+    withAsync (forever $ T.hGetLine stdO >>= customLogger stdO nt) $ \_ ->
+        withAsync (forever $ T.hGetLine stdE >>= customLogger stdE nt) $ \_ -> do
```

Comment:
Wait, does it even work? AFAIU, you are passing the same stream as the one you read from (`stdE`). You should pass `stderr` of our process (https://www.stackage.org/haddock/lts-9.1/base-4.9.1.0/System-IO.html#v:stderr). The same is true for `stdout`.

Reply:
Done. It worked before but it is not the output we are trying to get.